test/parallel/test-whatwg-url-properties.js

```diff
@@ -142,3 +142,19 @@ assert.strictEqual(url.searchParams, oldParams);
   assert.strictEqual(opts.search, '?l=24');
   assert.strictEqual(opts.hash, '#test');
 }
+
+// Test special origins
+[
+  { expected: 'https://whatwg.org',
```

Comment:
Specifically the `file:` origin test is not covered by the WPT, which would have uncovered the 'bug.' I just added other origin tests for completeness.